lua/telescope/pickers.lua

```diff
@@ -963,6 +1035,8 @@ function Picker:register_completion_callback(cb)
   table.insert(self._completion_callbacks, cb)
 end
 
+-- TODO: document what the purpose of this is
+--  Currently only used in `Picker: get_result_completor`
```

Comment:
completion callbacks can be registered via `opts.on_complete` basically functions that run after all items are proceeded once. Currently used in testing, but users could do anything with it.
and `_on_complete` is basically only calling all these registered callbacks